xir/program.py

```diff
@@ -137,27 +137,44 @@ def use_floats(self) -> bool:
         return self._use_floats
 
 
+class ObservableFactor:
+    def __init__(self, name: str, params: [], wires: [int]):
+        self.name = name
+        self.params = params
+        self.wires = wires
+
+    def __str__(self):
+        if len(self.params) == 0:
+            return f"{self.name}{self.wires}"
+        else:
+            return f"{self.name}({', '.join([str(v) for v in self.params])}){self.wires}"
+
+
 class ObservableStmt:
     """Observable statements to be used in observable definitions.
 
     Args:
         pref (Decimal, int, str): prefactor to the observable terms
-        terms (list): list of observables and the wire(s) they are applied to
+        factors (list): list of observables and the wire(s) they are applied to
```

Comment:
We should update the description here and replace "terms" with "factors" in the line above.
```suggestion
        factors (list[ObservableFactor]): list of observable factors
        use_floats(bool): whether floats and complex types are returned instead of ``Decimal``
```